### DISCUSSION TITLE
Prism Region Navigation for Maui Applications.

### DISCUSSION THREAD

**Discussion by maxxentropy** [PrismLibrary/Prism#2909] [.NET MAUI] [answered]
I have a couple of questions about Region navigation within Maui applications. I come from a windows/wpf perspective and have experience using Prism in this context. I am new to the mobile application development scene and am trying to learn Maui while taking advantage of what I know about Prism.

1.) There is no documentation or guidance for Region navigation in the Maui section of the documentation. Can we use documentation from the WPF section or has the Region navigation for Maui departed significantly enough from the original implementation to warrant a new documentation section and a deeper understanding of the Maui version of the source code?

2.) If I am focusing strictly on Mobile development ( no desktop functionality ) does using Region navigation even make sense? With a WPF application, I would typically use regions to describe a navigation section, content section, and other windows areas like toolbar region and the like for my App Shell. After reviewing the available Prism documentation, it seems like much of this can be managed with a tabbed view for navigation. There is not really an app shell with defined regions if we render a single page at a time, and navigate to it. Am I missing something?

**Comment by dansiegel** [comment, marked as answer]
> 1.) There is no documentation or guidance for Region navigation in the Maui section of the documentation. Can we use documentation from the WPF section or has the Region navigation for Maui departed significantly enough from the original implementation to warrant a new documentation section and a deeper understanding of the Maui version of the source code?

The documentation is effectively the same for both WPF and Maui. There are a few subtle differences currently but this is being worked on as we are looking to migrate the abstractions to Prism.Core for Prism 9.

> 2.) If I am focusing strictly on Mobile development ( no desktop functionality ) does using Region navigation even make sense? With a WPF application, I would typically use regions to describe a navigation section, content section, and other windows areas like toolbar region and the like for my App Shell. After reviewing the available Prism documentation, it seems like much of this can be managed with a tabbed view for navigation. There is not really an app shell with defined regions if we render a single page at a time, and navigate to it. Am I missing something?

This is ultimately a decision you need to make for yourself. Typical Xamarin.Forms applications (and by extension .NET MAUI) make primary use of a Page Navigation pattern. Region support is much newer on the Mobile side of things, however I have seen some really amazing things that developers have been able to do in their mobile apps since we've introduced it. So again this really boils down to what are you looking to accomplish and what sort of behavior do you want your user to have for instance when they click on the hardware back button. As an example if you simply launched a Shell the Maui app would have a single page which would result in the user backing out of the app any time they click the hardware back button.

**Reply by maxxentropy** [reply]
Thanks for the information Dan! 

If there are any developers out there that have examples of how to organize navigation using regions I would love to see them. I have a sample app that has a login view, sign up view, and a main view, defined as views of Modules, and the applications' main view is a ContentPage where the xaml is defined as

```
<?xml version="1.0" encoding="utf-8"?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:prism="http://prismlibrary.com"
             xmlns:mvvm="http://prismlibrary.com"
             mvvm:ViewModelLocator.AutowireViewModel="Automatic"
             x:Class="VLink.Mobile.Views.ContentRegionPageView">
  <ContentPage.Content>
    <ContentView prism:RegionManager.RegionName="ContentRegion" />
  </ContentPage.Content>
</ContentPage> 
```   

I am just pushing the views into the single content region using the Navigation service. I don't know if this is a reasonable thing to do, or if I am going to run into issues down the road that will make this approach untenable.
